### DISCUSSION TITLE
A way to supply a link and commentary

### DISCUSSION THREAD

**Discussion by bentsai** [neurosnap/lists.sh#25] [General]
I find myself wanting to link to a URL and write a comment around it. One use case being, I find a cool link and I add it to my list, and I also want to talk about why it's awesome. Here's an example:

https://ben.lists.sh/2022-05-17-daily-note

Using the existing mechanism of link text doesn't make sense (`=> <url> <linktext>`) because the entire comment might not apply to the link, and also it obscures the URL.

The second list item I have is almost what I want, but the URL doesn't turn into an anchor tag.

I guess auto-linking detected URLs might solve this. Another thought is, if you have nested lists, you could have the top level be the URL, and then nest the description under it. I wonder if you might special case that as [a HTML description list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)? Just musing here, I'm not proposing any solutions :)

**Comment by neurosnap** [comment]
I think having a nested list might be the solution here.  I don't want to parse the entire text body in order to render links and I don't want to include an javascript on the FE.

My current plan is to support a single level of nesting and that's it in order to keep parsing to the strict "look ahead 3 characters per line to figure out the type."

**Reply by neurosnap** [reply]
#26
